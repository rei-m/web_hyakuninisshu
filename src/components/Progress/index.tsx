import * as React from 'react';
import styled from 'styled-components';
import { Classes, Spinner } from '@blueprintjs/core';
import { lifecycle } from 'recompose';

export interface ProgressProps {
  readonly onStart: () => void;
}

const Root = styled.div`
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fffff050;
  flex-direction: column;
  height: 100vh;
  position: fiexd;
`;

const Message = styled.div`
  margin-top: 16px;
  font-size: 2rem;
`;

const Progress: React.SFC<ProgressProps> = _props => (
  <Root>
    <Spinner className={Classes.LARGE} />
    <Message>　　　百人一首 準備中。。。</Message>
  </Root>
);

export default lifecycle<ProgressProps, {}>({
  componentDidMount() {
    this.props.onStart();
  }
})(Progress);
