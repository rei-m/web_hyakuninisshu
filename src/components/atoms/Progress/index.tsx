import * as React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { appTheme } from '@src/styles/theme';

export interface Props {
  size?: number;
  color?: string;
}

const Progress = ({ size = 64, color = appTheme.colorPrimaryDark }: Props) => (
  <CircularProgress style={{ color, width: size, height: size }} />
);

export default Progress;
