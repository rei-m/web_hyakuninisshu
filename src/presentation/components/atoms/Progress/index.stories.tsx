import React from 'react';
import { storiesOf } from '@storybook/react';
import Progress from './index';

storiesOf('atoms/Progress', module)
  .add('default', () => <Progress />)
  .add('custom', () => <Progress size={200} color={`#0f0`} />);
