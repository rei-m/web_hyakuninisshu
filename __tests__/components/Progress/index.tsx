import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import Progress from '../../../src/components/Progress';

describe('<Progress />', () => {
  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<Progress onStart={jest.fn()} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
