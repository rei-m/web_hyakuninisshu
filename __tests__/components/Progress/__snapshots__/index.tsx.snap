// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> should render component 1`] = `
<div
  className="sc-bdVaJa fmvnED"
>
  <div
    className="bp3-spinner bp3-large"
  >
    <svg
      height={100}
      strokeWidth={4}
      viewBox="0 0 100 100"
      width={100}
    >
      <path
        className="bp3-spinner-track"
        d="M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1 1 0,-89"
      />
      <path
        className="bp3-spinner-head"
        d="M 50,50 m 0,-44.5 a 44.5,44.5 0 1 1 0,89 a 44.5,44.5 0 1 1 0,-89"
        pathLength={280}
        strokeDasharray="280 280"
        strokeDashoffset={210}
      />
    </svg>
  </div>
  <div
    className="sc-bwzfXH jeCWiS"
  >
    　　　百人一首 準備中。。。
  </div>
</div>
`;
